src/store/products/products.hooks.ts

```diff
@@ -0,0 +1,44 @@
+import axios, { AxiosResponse } from "axios";
+
+import { useApiFetch, useApiMutate } from "../api.hooks";
+import {
+  Product,
+  ProductsFetchResponseData,
+  PRODUCTS_FETCH_SUCCESS,
+  ProductsAddParams,
+  ProductsAddResponseData,
+  PRODUCTS_ADD_SUCCESS
+} from "./products.types";
+import { getProducts } from "./products.selectors";
+
+export const useFetchProducts = () =>
+  useApiFetch({
+    requestFn: async () => {
+      const res: AxiosResponse<ProductsFetchResponseData> = await axios.get(
+        `${process.env.REACT_APP_API_URL}/products`
+      );
+
+      return res.data.products;
+    },
+    cacheActionCreator: (data: Product[]) => ({
+      type: PRODUCTS_FETCH_SUCCESS,
+      payload: data
+    }),
+    cacheSelector: getProducts
+  });
```

Comment:
this I haven't seen before, is this caching the fetched data so that it won't need re-fetched every time it's viewed?

Reply:
I agree, better to learn the hard way than take shortcuts.